### PR TITLE
Only build compatjit.dll when building JIT32.

### LINKED ISSUE #10106: JIT: only include compatjit.dll in JIT package for JIT32
`.nuget\Microsoft.NETCore.Jit\runtime.Windows_NT.Microsoft.NETCore.Jit.props` contains the following line that causes it to include compatjit.dll in the JIT NuGet package:

```
<NativeBinary Condition="'$(Platform)' == 'x86'" Include="$(BinDir)compatjit.dll" />
```

There should be one additional condition: only do this if we are building JIT32. In CMakeLists.txt, this is true with the `CLR_BUILD_JIT32` boolean is true.

If the NuGet package build is conditionalized, then the following in src\jit\CMakeLists.txt can be removed (thus removing one JIT build flavor):
```
if (CLR_CMAKE_PLATFORM_ARCH_I386 AND WIN32)
    if (NOT CLR_BUILD_JIT32)
        add_subdirectory(compatjit)
    endif ()
endif (CLR_CMAKE_PLATFORM_ARCH_I386 AND WIN32)
```

Also, the src\jit\compatjit directory and its contents can be removed.

### PR DESCRIPTION
**Pull request by pgavlin**
In order to avoid confusion, do not build and package compatjit.dll
unless we are building JIT32.

Fixes #10106.

### CODE REVIEW

**Comment by pgavlin** [comment]
@BruceForstall @RussKeldorph @dotnet/jit-contrib PTAL

**Comment by BruceForstall** [comment]
There's a bunch of compatjit stuff in netci.groovy (using architecture "x86compatjit") that needs to be deleted, or else presumably some periodically-invoked legs will fail (as well as if you trigger them). You can probably also delete tests\x86\compatjit_x86_testenv.cmd. Presumably internal JIT32 testing will handle invoking the JIT32 fallback path properly.

File: build.cmd [hunk 1]
```diff
@@ -409,6 +409,10 @@ if %__BuildCoreLib% EQU 1 (
 		set __nugetBuildArgs=-buildNugetPackage=true
 	)
 
+    set PackageCompatJit=
+    if "%__BuildJit32%" == "1" (
+        set PackageCompatJit=1
+    )
     @call %__ProjectDir%\run.cmd build -Project=%__ProjectDir%\build.proj -MsBuildLog=!__MsbuildLog! -MsBuildWrn=!__MsbuildWrn! -MsBuildErr=!__MsbuildErr! !__nugetBuildArgs! %__RunArgs% %__UnprocessedBuildArgs%
```
Comment (BruceForstall) [inline build.cmd]: Can you please add a comment about why we need this, and where it is used?

**Comment by BruceForstall** [comment]
The build-related changes look fine to me, as long as they work :)

**Comment by danmosemsft** [comment]
@pgavlin  where are we at on this one..

**Comment by pgavlin** [comment]
@danmosemsft: I still need to make the required changes to our groovy scripts.

**Comment by pgavlin** [comment]
@BruceForstall: I updated the groovy scripts; PTAL

**Comment by pgavlin** [comment]
@dotnet-bot test ci

**Comment by BruceForstall** [comment]
Yay for simplifying netci.groovy!

You'll want to make sure the mirror doesn't choke on removing a mirrored directory (and make sure the compatjit directory gets deleted from TFS)

**Comment by adiaaida** [comment]
@dotnet-bot test Windows_NT x86jit32 perf please

**Comment by pgavlin** [comment]
@dotnet-bot test ci

**Comment by pgavlin** [comment]
@dotnet-bot test ci

**Comment by adiaaida** [comment]
@dotnet-bot test Windows_NT x86jit32 perf

**Comment by pgavlin** [comment]
@dotnet-bot test ci

**Comment by pgavlin** [comment]
@dotnet-bot test ci

**Pull request merged**